src/jsifier.js

```diff
@@ -266,14 +266,12 @@ function JSify(data, functionsOnly) {
         // Emit the body of a JS library function.
         var proxyingMode = LibraryManager.library[ident + '__proxy'];
         if (USE_PTHREADS && proxyingMode) {
-          var sig = LibraryManager.library[ident + '__sig'];
-          if (!sig) throw 'Missing function signature field "' + ident + '__sig"! (Using proxying mode requires specifying the signature of the function)';
-          sig = sig.replace(/f/g, 'i'); // TODO: Implement float signatures.
           if (proxyingMode !== 'sync' && proxyingMode !== 'async') {
             throw 'Invalid proxyingMode ' + ident + '__proxy: \'' + proxyingMode + '\' specified!';
           }
           var sync = proxyingMode === 'sync';
-          if (sig.length > 1) {
+          var hasArgs = snippet.match(/function\s+.*?\s*\((.*?)\)\s*{/)[1].length > 0;
```

Comment:
Why change this? (This will match `function( )` and `function(/*this does not need args*/)` to have arguments)

If this needs to be special-cased for variadic functions, then perhaps we could declare syscalls with `__syscall142__sig: ['i*']`, and slap that to each syscall?

Reply:
Also looks like below we could just give syscalls signatures to not have to regex here?